tsp-output/@azure-tools/typespec-csharp/src/Generated/Models/CreateTranscriptionRequest.Serialization.cs

```diff
@@ -2,191 +2,63 @@
-                if (property.NameEquals("language"u8))
-                {
-                    language = property.Value.GetString();
-                    continue;
-                }
-                if (property.NameEquals("prompt"u8))
-                {
-                    prompt = property.Value.GetString();
-                    continue;
-                }
-                if (property.NameEquals("response_format"u8))
-                {
-                    if (property.Value.ValueKind == JsonValueKind.Null)
-                    {
-                        continue;
-                    }
-                    responseFormat = new CreateTranscriptionRequestResponseFormat(property.Value.GetString());
-                    continue;
-                }
-                if (property.NameEquals("temperature"u8))
-                {
-                    if (property.Value.ValueKind == JsonValueKind.Null)
-                    {
-                        continue;
-                    }
-                    temperature = property.Value.GetDouble();
-                    continue;
-                }
-                if (options.Format != "W")
-                {
-                    additionalPropertiesDictionary.Add(property.Name, BinaryData.FromString(property.Value.GetRawText()));
-                }
+                // TODO: preferred way to handle floats/numerics?
+                content.Add(new StringContent($"{Temperature}"), "temperature");
             }
-            serializedAdditionalRawData = additionalPropertiesDictionary;
-            return new CreateTranscriptionRequest(file, model, language.Value, prompt.Value, OptionalProperty.ToNullable(responseFormat), OptionalProperty.ToNullable(temperature), serializedAdditionalRawData);
-        }
 
-        BinaryData IPersistableModel<CreateTranscriptionRequest>.Write(ModelReaderWriterOptions options)
-        {
-            var format = options.Format == "W" ? ((IPersistableModel<CreateTranscriptionRequest>)this).GetFormatFromOptions(options) : options.Format;
-
-            switch (format)
+            
+            string contentType = default;
+            if (content.Headers.ContentType is MediaTypeHeaderValue contentTypeValue)
             {
-                case "J":
-                    return ModelReaderWriter.Write(this, options);
-                default:
-                    throw new FormatException($"The model {nameof(CreateTranscriptionRequest)} does not support '{options.Format}' format.");
+                contentType = contentTypeValue.ToString();
             }
-        }
 
-        CreateTranscriptionRequest IPersistableModel<CreateTranscriptionRequest>.Create(BinaryData data, ModelReaderWriterOptions options)
-        {
-            var format = options.Format == "W" ? ((IPersistableModel<CreateTranscriptionRequest>)this).GetFormatFromOptions(options) : options.Format;
-
-            switch (format)
+            // TODO: transfer all headers instead of a few?
```

Comment:
TODO: Validate that Content-Type is the only one strictly required on the top-level request and that all other headers are either a) ok to leave off or b) written inline in the multipart content payload.